pkg/trace/api/api.go

```diff
@@ -357,6 +367,48 @@ func (r *HTTPReceiver) rateLimited(n int64) bool {
 	return !r.RateLimiter.Permits(n)
 }
 
+// StatsProcessor implementations are able to process incoming client stats.
+type StatsProcessor interface {
+	// ProcessStats takes a stats payload and consumes it. It is considered to be originating
+	// from the given lang.
+	ProcessStats(p pb.ClientStatsPayload, lang string)
+}
+
+// handleStats handles incoming stats payloads.
+func (r *HTTPReceiver) handleStats(w http.ResponseWriter, req *http.Request) {
+	defer timing.Since("datadog.trace_agent.receiver.stats_process_ms", time.Now())
+
+	rd := NewLimitedReader(req.Body, r.conf.MaxRequestBytes)
+	slurp, err := ioutil.ReadAll(rd)
+	if err != nil {
+		httpDecodingError(err, []string{"handler:stats", "v:v0.5"}, w)
+		return
+	}
+	// TODO(gbbr): add tagStats to all metrics
+	metrics.Count("datadog.trace_agent.receiver.stats_payload", 1, nil, 1)
+	metrics.Count("datadog.trace_agent.receiver.stats_bytes", int64(len(slurp)), nil, 1)
+
+	req.Header.Set("Accept", "application/msgpack")
+	req.Header.Add("Accept", "application/protobuf")
+
+	var in pb.ClientStatsPayload
+	switch ct := getMediaType(req); ct {
+	case "application/msgpack":
+		if err := msgp.Decode(bytes.NewReader(slurp), &in); err != nil {
+			httpDecodingError(err, []string{"handler:stats", "codec:msgpack", "v:v0.5"}, w)
+			return
+		}
+	default:
+		if err := proto.Unmarshal(slurp, &in); err != nil {
```

Comment:
I think we should now cleanup proto support as it's implemented with msgpack

Reply:
Done.